doc/upgrading.md

```diff
@@ -0,0 +1,40 @@
+# Managing Upgrades/Downgrades
+This guide describes how to upgrade your SPIRE deployment, as well as the compatibility guarantees that SPIRE users can expect.
+
+## SPIRE Versioning
+SPIRE versions are expressed as **x.y.z**, where **x** is the major version, **y** is the minor version, and **z** is the patch version, following Semantic Versioning terminology. The SPIRE project is currently pre-1.0, so minor SPIRE releases act like major releases and patch releases act like minor releases. Despite the fact that the project is pre-1.0, version compatibility guarantees are already in place.
+
+### SPIRE Server Compatibility
+Version skew within a SPIRE server cluster is supported within +/- 1 minor version. In other words, the newest and oldest SPIRE server instances in any given cluster must be within one minor version of each other.
+
+For example:
+* Newest SPIRE server instance is at 0.9.3
+* Other SPIRE server instances are supported at 0.9.x and 0.8.x
+
+### SPIRE Agent Compatibility
+SPIRE agents must not be newer than the oldest SPIRE server that they communicate with, and may be up to one minor version older.
+
+For example:
+* SPIRE servers are at both 0.9.3 and 0.9.2
+* SPIRE agents are supported at 0.8.0 through 0.9.2
+
+## Supported Upgrade Paths
+
+The supported version skew between SPIRE servers and agents has implications on the order in which they must be upgraded. SPIRE servers must be upgraded before SPIRE agents, and is limited to a jump of at most one minor version (regardless of patch version). SPIRE server and agent instances may be upgraded in a rolling fashion.
+
+For example, if upgrading from 0.8.1 to 0.9.3:
+* Upgrade SPIRE server instances from 0.8.1 to 0.9.3 one at a time
```

Comment:
the "one at a time" phrasing is a bit ambiguous. one _instance_ at a time? one _minor_ version at a time?

the phrasing occurs in a few spots...

Reply:
Would this wording address your concern?

```
Upgrade SPIRE servers from 0.8.1 to 0.9.3 one instance at a time
```

I should add that I went back and forth on whether we should include the rollout methodology on this line or not... the goal is to upgrade from one to the other, any ordering/batching etc will accomplish that, but there's one that we'd recommend over others... wasn't quite sure how/where to capture that in a way that it wouldn't be overlooked

Reply:
Yeah, that's a fair concern. Maybe we're being too prescriptive... maybe we just say something along the lines of what you just said.

Reply:
Fixed